examples/data-objects/focus-tracker/src/FocusTracker.ts

```diff
@@ -53,11 +51,22 @@ export class FocusTracker extends DataObject<{}, undefined, IFocusTrackerEvents>
         }
         return this._audience;
     }
-    public set audience(newAudience: IServiceAudience<IMember>) {
-        if (this._audience !== undefined) {
-            throw new Error("set audience only once");
+
+    private _signalManager: ISignalManager | undefined;
+    public get signalManager(): ISignalManager {
+        if (this._signalManager === undefined) {
+            throw new Error("no signalManager");
+        }
+        return this._signalManager;
+    }
+
+    public init(newAudience: IServiceAudience<IMember>, newSignalManager: ISignalManager) {
+        if (this._audience !== undefined || this._signalManager !== undefined) {
+            throw new Error("init only once");
         }
         this._audience = newAudience;
+        this._signalManager = newSignalManager;
```

Comment:
If we're considering moving `SignalManager` to be a class instead of a data object, we could have the `init` function here pass `FocusTracker`'s runtime over to the constructor for the new `SignalManager` and instantiate it in here instead of having it be passed in

Reply:
Yep that's roughly where I'm at with changing it to a class.  This works well for custom DOs, but we're stuck with the fluid-static v1 case where custom DOs aren't expected :/